plugins/woocommerce/src/Blocks/BlockTypes/AbstractBlock.php

```diff
@@ -275,6 +298,43 @@ function ( $html, $block ) use ( $style_handles ) {
 		);
 	}
 
+	/**
+	 * Transform block metadata to the expected format.
+	 *
+	 * @param array $block_meta The original block metadata.
+	 * @return array Transformed block metadata.
+	 */
+	protected function transform_block_metadata( $block_meta ) {
+		$property_mappings = array(
+			'apiVersion'      => 'api_version',
+			'name'            => 'name',
+			'title'           => 'title',
+			'category'        => 'category',
+			'parent'          => 'parent',
+			'ancestor'        => 'ancestor',
+			'icon'            => 'icon',
+			'description'     => 'description',
+			'keywords'        => 'keywords',
+			'attributes'      => 'attributes',
+			'providesContext' => 'provides_context',
+			'usesContext'     => 'uses_context',
+			'selectors'       => 'selectors',
+			'supports'        => 'supports',
+			'styles'          => 'styles',
+			'variations'      => 'variations',
+			'example'         => 'example',
+			'allowedBlocks'   => 'allowed_blocks',
+		);
```

Comment:
Why can't this be done at build time?

Reply:
Good idea. I have moved it to the build time: ee1e1226fd654c9723ca9318678da5c941b954c1